suricata/update/main.py

```diff
@@ -608,10 +608,13 @@ def write_merged(filename, rulemap):
                         len(report["added"]),
                         len(report["removed"]),
                         len(report["modified"])))
-    
-    with io.open(filename, encoding="utf-8", mode="w") as fileobj:
+
+    with io.open(tmp_file.name, encoding="utf-8", mode="w") as fileobj:
         for rule in rulemap:
             print(rulemap[rule].format(), file=fileobj)
+    # Copying the rule file from /tmp to the actual destination
+    shutil.copy(tmp_file.name, filename)
+    tmp_file.close()
 
```

Comment:
After thinking about this a bit, I think we can do without the temporary directory and copy. For this case we are outputing a single file, to a directory that we can assume we have write access to, and that file name is always suricata.rules as we hard code it for now. So we could in:
- use "suricata.rules.tmp" as a the temp file, in the same directory as the final file.
- then os.rename suricata.rules.tmp to suricata.rules
which should be faster, and safer.

Something similar could probably be done for the write_to_directory. But I'm not exactly sure how that would look at the moment. Needs a bit more thought I think.

Reply:
About `write_to_directory`, since those files also do have a definite name, we could use the same scheme, I think.